Geras is Back

hivehum opened the pull request:
draft bc im getting a lot of feedback that isnt addressed yet. this is literally just a revert of space-wizards/space-station-14#30912 and space-wizards/space-station-14#29731 with as of writing no actual mechanical changes, which it likely needs bc while slimes are vulnerable to water damage now they can also vent crawl which makes this a crazy tactical advantage

edit: geras can no longer vent crawl. you can see the interior of a slime in both forms on examine. i think firestack and zombification immunity & losing wanted status have been fixed, im still testing

https://github.com/user-attachments/assets/37463ad1-2050-4da6-8ad3-5454921c1bb1

to do list

- [X]  geras specific sprite?
- [X] remove vent crawling
- [X] address most of the open issues regarding geras before they were removed

**Changelog**
:cl:
- add: After months of censorship, slimes are once again able to let it all hang out.
- add: You can now see inside slime people.

RosySaturniidae commented:
i would rather keep internal storage being kept on transform than give them vent crawling, if only because any other source of vent crawling for people is locked behind antagonist activities, the syndicate contortionist suit and the changeling lesser form (? that might have been changed i dont remember). being able to bypass all access, at any time, is fairly crazy, even if you do remove all items from their person as a result. 

meanwhile, slime storage is a fairly niche ability that pretty much only comes up if youre trying to hide contraband or if someone is trying to bomb you. (storage isnt as much of a premium here than it is on some other servers, so the extra 2x3 isnt used thaaat often) it would be nice to add an additional use for it.

Darkmajia commented:
worth noting geras likely still has all these issues that were never touched upstream
![image](https://github.com/user-attachments/assets/2f7fab6a-6e74-40ef-910d-150d6d8560ac)

hivehum commented:
> I'll admit, I'm not in favour of this, and would rather slimes stay as-is. Their internal storage is plentifully convenient, despite Rosy remarking on storage creep - and their high resistance to atmos loss is a key element of the species at this point. Why do they need more perks?

i think you're really overestimating the strategic advantage of geras form. having to drop literally everything except what fits in a 2x3 pocket to become the weakest vent critter in the game. you trade 120 of your max hp (geras does not have a crit form and goes straight to dead at 80) for a move speed increase. if i remove vent crawling (which i am leaning towards now because of feedback) then you are not even able to leave whatever room you morph in

ive really always adored geras as a flavor addition to slime people. i think we should generally strive to make the upstream playable species as mechanically unique as our species and restoring geras makes slimes significantly different from humans rather than it being almost entirely a matter of damage resistances

hivehum commented:
geras updates: added code from space-wizard/space-station-14#27417 to make geras form actually match your slime color, added smile's face to the sprite so you look as visually distinct from a vent slime as is reasonable. i dont really want to deal with adding actual geras customization because thats probably going to be character menu stuff. i want to focus on geras being in a usable state and everything else can come later

![SS14 Loader_jNQPIVZP9Q](https://github.com/user-attachments/assets/de75f6e7-dbe9-4510-911f-46e88d42a546)
![SS14 Loader_kOZNbQ38x6](https://github.com/user-attachments/assets/364019ef-9a92-4d06-9b0b-4e84781981dd)
![SS14 Loader_EGc3JAhMZ8](https://github.com/user-attachments/assets/2dae06d2-4760-4b51-b0bf-fc967f432f28)

Pinkbat5 commented:
Since a lot of slime characters don't have mouths, how does the face look without the little 3 mouth? Do the eyes still read as eyes?
Cool as hell feature, glad to see it returning.

nowaitstop commented:
> > I'll admit, I'm not in favour of this, and would rather slimes stay as-is. Their internal storage is plentifully convenient, despite Rosy remarking on storage creep - and their high resistance to atmos loss is a key element of the species at this point. Why do they need more perks?
> 
> i think you're really overestimating the strategic advantage of geras form. having to drop literally everything except what fits in a 2x3 pocket to become the weakest vent critter in the game. you trade 120 of your max hp (geras does not have a crit form and goes straight to dead at 80) for a move speed increase. if i remove vent crawling (which i am leaning towards now because of feedback) then you are not even able to leave whatever room you morph in

Just saw this PR;  maybe a happy medium is a slime's cavity emptying when going geras mode as a trade off for being able to vent crawl... best used for escaping dangerous situations (including closed rooms), not so much for crimes other than secure trespass since a slime wouldnt be able to enter with a weapon OR leave with stolen goods

hivehum commented:
took a lot of code from space-wizard/space-station-14#28941 which i hope will fix most of the outstanding issues w geras (and polymorphing in general). also it turns out getting vent crawling out of geras specifically was really easy. 

i want to make geras real i miss her so bad

mqole commented:
i dunno what i did but i guess it worked. sorry about the commits

hivehum commented:
mq did something truly terrible here. remaking the PR

mqole commented:
![image](https://github.com/user-attachments/assets/843e2ea5-7209-402e-ba87-a7c9792fe167)
